src/dodal/devices/electron_analyser/abstract_analyser_io.py

```diff
@@ -25,13 +40,81 @@ def __init__(self, prefix: str, name: str = "") -> None:
             self.energy_step = epics_signal_rw(float, prefix + "STEP_SIZE")
             self.iterations = epics_signal_rw(int, prefix + "NumExposures")
             self.acquisition_mode = epics_signal_rw(str, prefix + "ACQ_MODE")
+            self.step_time = epics_signal_r(float, prefix + "AcquireTime")
+
+            self.total_steps = self._create_total_steps_signal(prefix)
+            self.total_time = derived_signal_r(
+                self._calculate_total_time,
+                "s",
+                total_steps=self.total_steps,
+                step_time=self.step_time,
+                iterations=self.iterations,
+            )
+
+        with self.add_children_as_readables():
+            self.image = epics_signal_r(Array1D[np.float64], prefix + "IMAGE")
+            self.spectrum = epics_signal_r(Array1D[np.float64], prefix + "INT_SPECTRUM")
+            self.total_intensity = derived_signal_r(
+                self._calculate_total_intensity, spectrum=self.spectrum
+            )
+            # ToDo - Ideally the below are only collected once per region. However, they
+            # need to be read after the first point of a region (stream). Bluesky /
+            # ophyd currently doesn't support this and therefore must be read per point
+            # as a workaround, otherwise they return the previous region values.
+            self.excitation_energy = soft_signal_rw(float, initial_value=0, units="eV")
+            self.energy_axis = self._create_energy_axis_signal(prefix)
+            self.binding_energy_axis = derived_signal_r(
+                self._calculate_binding_energy_axis,
+                "eV",
+                energy_axis=self.energy_axis,
+                excitation_energy=self.excitation_energy,
+                energy_mode=self.energy_mode,
+            )
+            self.angle_axis = self._create_angle_axis_signal(prefix)
+        super().__init__(prefix=prefix, name=name)
+
+    @abstractmethod
+    def _create_angle_axis_signal(self, prefix: str) -> SignalR[Array1D[np.float64]]:
+        """
+        The signal that defines the angle axis. Depends on analyser model.
+        """
+
+    @abstractmethod
+    def _create_energy_axis_signal(self, prefix: str) -> SignalR[Array1D[np.float64]]:
+        """
+        The signal that defines the energy axis. Depends on analyser model.
+        """
 
-        super().__init__(name)
+    def _calculate_binding_energy_axis(
+        self,
+        energy_axis: Array1D[np.float64],
+        excitation_energy: float,
+        energy_mode: EnergyMode,
+    ) -> Array1D[np.float64]:
+        is_binding = energy_mode == EnergyMode.BINDING
+        return np.array(
+            [
+                to_binding_energy(i_energy_axis, EnergyMode.KINETIC, excitation_energy)
+                if is_binding
+                else i_energy_axis
+                for i_energy_axis in energy_axis
+            ]
+        )
```

Comment:
```suggestion
        is_binding = energy_mode == EnergyMode.BINDING
        return np.array(
            [to_binding_energy(i_energy_axis, energy_mod, excitation_energy) for i_energy_axis in energy_axis]
        )
```

Isn't this the same?

Reply:
No, tests fail if you do it this way. The reason being is we actually need to give it the opposite of the energy_mode supplied. For example, if it is already in binding energy, the pv returns the energy axis to us as kinetic energy and therefore we need to convert it to binding energy.